scripts/n2k/collection.py

```diff
@@ -0,0 +1,169 @@
+        "gdb": f"Natura 2000 {year} Land Cover/Land Use Status {file_format}",
+        "xml": f"Natura 2000 {year} Land Cover/Land Use Status Metadata",
+        "lyr": f"Natura 2000 {year} Land Cover/Land Use Status ArcGIS Layer File",
+        "qml": f"Natura 2000 {year} Land Cover/Land Use Status QGIS Layer File",
+        "sld": f"Natura 2000 {year} Land Cover/Land Use Status OGC Layer File",
+    }
+    role_map = {
+        "zip": ["data"],
+        "gpkg": ["data"],
+        "gdb": ["data"],
+        "xml": ["metadata"],
+        "lyr": ["metadata"],
+        "qml": ["metadata"],
+        "sld": ["metadata"],
+    }
+    return asset_id, pystac.Asset(
+        href=asset_path, media_type=media_type_map[extension], title=title_map[extension], roles=role_map[extension]
+    )
+
+
+def collect_assets(n2k_root: str) -> list[str]:
+    asset_list = (
+        get_files(n2k_root, "xml")
+        + get_files(n2k_root, "lyr")
+        + get_files(n2k_root, "qml")
+        + get_files(n2k_root, "sld")
+        + get_files(n2k_root, "gpkg")
+        + get_gdb(n2k_root)
+        + get_files(n2k_root, "zip")
+    )
+    assets = {}
+    for asset_path in asset_list:
+        _, tail = os.path.split(asset_path)
+        asset_id, asset = create_asset(tail, asset_path)
+        if asset_id not in assets:
+            assets[asset_id] = asset
+    return assets
+
+
+def add_summaries_to_collection(collection: pystac.Collection, epsg_list: list[int]) -> None:
+    summaries = ProjectionExtension.summaries(collection, add_if_missing=True)
+    summaries.epsg = epsg_list
+
+
+def add_links_to_collection(collection: pystac.Collection, link_list: list[pystac.Link]) -> None:
+    for link in link_list:
+        collection.links.append(link)
+
+
+def add_assets_to_collection(collection: pystac.Collection, asset_dict: dict[str, pystac.Asset]) -> None:
+    for key, asset in asset_dict.items():
+        collection.add_asset(key, asset)
+
+
+def create_collection(n2k_root: str) -> pystac.Collection:
+    try:
+        collection = pystac.Collection(
+            id=COLLECTION_ID,
+            description=COLLECTION_DESCRIPTION,
+            extent=COLLECTION_EXTENT,
+            title=COLLECTION_TITLE,
+            keywords=COLLECTION_KEYWORDS,
+            providers=[N2K_HOST_AND_LICENSOR],
+        )
+
+        # summaries
+        epsg_list = [3035]
+        add_summaries_to_collection(collection, epsg_list)
+
+        # links
+        link_list = [CLMS_LICENSE]
+        add_links_to_collection(collection, link_list)
+
+        # assets
+        assets = collect_assets(n2k_root)
+        add_assets_to_collection(collection, assets)
+
+        # update links
+        collection.set_self_href(os.path.join(WORKING_DIR, f"{STAC_DIR}/{COLLECTION_ID}/{collection.id}.json"))
+        catalog = pystac.read_file(f"{WORKING_DIR}/{STAC_DIR}/clms_catalog.json")
+        collection.set_root(catalog)
+        collection.set_parent(catalog)
+    except Exception as error:
```

Comment:
I think is ok to define a custom exception but i would still catch concrete errors. You can list everything in the same line (i don't know which could be exceptions) Something like this 

except (OSError, pystac.STACError, list here the errors) as error:
    raise CollectionCreationError(f"Error during collection creation: {error}")